Source/WebCore/dom/Node.h

```diff
@@ -421,6 +421,7 @@ class Node : public EventTarget {
     // Integration with rendering tree
 
     // As renderer() includes a branch you should avoid calling it repeatedly in hot code paths.
+    // Make sure to call `InspectorInstrumentation::didChangeRendererForDOMNode` immediately after.
     RenderObject* renderer() const { return m_rendererWithStyleFlags.pointer(); }
     void setRenderer(RenderObject*); // Defined in RenderObject.h
```

Comment:
Why doesn't setRenderer call InspectorInstrumentation::didChangeRendererForDOMNode instead?

Reply:
i'd tried this at first, but got a whole bunch of linker errors so i went with another approach just to see if i could get something working at all

i took another look today and think ive fixed all the issues :)